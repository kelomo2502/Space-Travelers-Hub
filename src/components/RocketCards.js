import PropTypes from 'prop-types';
import React from 'react';
import { useDispatch } from 'react-redux';
import { reservedRocket, cancelledReserve } from '../redux/rockets/rockets';

const RocketCards = ({
  image,
  title,
  description,
  rocketId,
  isReserved,
}) => {
  const dispatch = useDispatch();

  const handleReservation = (e) => {
    const { id } = e.target;
    dispatch(reservedRocket(id));
  };

  const handleCancelReservation = (e) => {
    const { id } = e.target;
    dispatch(cancelledReserve(id));
  };

  return (
    <div className="Rocket-card">
      <img src={image} alt="Space rockets" />
      <div>
        <h4>{title}</h4>
        <p className="Description">
          {isReserved ? <span className="Badge">Reserved</span> : ''}
          {description}
        </p>
        {isReserved
          ? (
            <button
              className="Cancel-btn"
              type="button"
              id={rocketId}
              onClick={handleCancelReservation}
            >
              Cancel Reservation
            </button>
          )
          : (
            <button
              className="Reserve-btn"
              type="button"
              id={rocketId}
              onClick={handleReservation}
            >
              Reserve Rocket
            </button>
          )}
      </div>
    </div>
  );
};

RocketCards.propTypes = {
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  rocketId: PropTypes.number.isRequired,
  isReserved: PropTypes.bool.isRequired,
};

export default RocketCards;
