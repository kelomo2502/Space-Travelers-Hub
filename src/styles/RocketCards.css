.Rocket-card {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  padding: 0 38px;
}

.Rocket-card img {
  width: 200px;
  height: 150px;
  margin: 18px;
}

.Rocket-card p {
  width: 70%;
}

.Reserve-btn {
  padding: 6px 14px;
  border: none;
  border-radius: 4px;
  background-color: #007bff;
  color: #fff;
}

.Reserve-btn:hover {
  cursor: pointer;
  background-color: #fff;
  color: #007bff;
  border: 1px solid #007bff;
}

.Cancel-btn {
  padding: 6px 14px;
  border: 1px solid #007bff;
  border-radius: 4px;
  background-color: #fff;
}

.Cancel-btn:hover {
  cursor: pointer;
  background-color: #007bff;
  color: #fff;
  border: none;
}

.Description {
  line-height: 24px;
}

.Badge {
  margin: 0 8px 8px 0;
  padding: 4px 8px;
  background-color: #00a2b8;
  color: #fff;
  border-radius: 4px;
}
