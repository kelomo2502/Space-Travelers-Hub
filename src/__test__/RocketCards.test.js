import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configurestore';
import RocketCards from '../components/RocketCards';

it('Renders Rockets component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <RocketCards />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
