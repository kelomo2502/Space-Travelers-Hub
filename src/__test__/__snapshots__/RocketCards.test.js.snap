// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Rockets component correctly 1`] = `
<div
  className="Rocket-card"
>
  <img
    alt="Space rockets"
  />
  <div>
    <h4 />
    <p
      className="Description"
    />
    <button
      className="Reserve-btn"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
